biosys/apps/main/utils_data_package.py

```diff
@@ -472,28 +472,59 @@ def get_fk_for_model(self, model_name):
                 return fk
         return None
 
+    def has_fk_for_model_field(self, model_name, model_field):
+        return self.get_fk_for_model_field(model_name, model_field) is not None
+
+    def get_fk_for_model_field(self, model_name, model_field):
+        for fk in self.foreign_keys:
+            if fk.model == model_name and fk.model_field == model_field:
+                return fk
+        return None
+
     def __str__(self):
         return self.get('name')
 
 
 class ObservationSchema(GenericSchema):
     """
-    A schema specific to an Observation Dataset.
-    It's main job is to deal with the observation date and it's geometry
-    (lat/long or geojson)
+     A schema specific to an Observation Dataset.
+     It's main job is to deal with the observation date and it's geometry
+     (lat/long or geojson)
+     There's a special case: a lat/long or geometry field can be omitted if there's a reference (foreign key)
+     to a site code (only)
     """
     OBSERVATION_DATE_FIELD_NAME = 'Observation Date'
     LATITUDE_FIELD_NAME = 'Latitude'
     LONGITUDE_FIELD_NAME = 'Longitude'
     DATUM_FIELD_NAME = 'Datum'
+    SITE_CODE_FOREIGN_KEY_EXAMPLE = """
+        "foreignKeys": [
+            {
+                "fields": ["Site Code"],
+                "reference": {
+                    "fields": ["code"],
+                    "resource": "Site"
+                }
+            }
+        ]
+     """
 
     def __init__(self, schema):
-        GenericSchema.__init__(self, schema)
+        super(ObservationSchema, self).__init__(schema)
         self.observation_date_field = self.find_observation_date_field_or_throw(self)
-        self.latitude_field = self.find_latitude_field_or_throw(self)
-        self.longitude_field = self.find_longitude_field_or_throw(self)
-        self.datum_field = self.find_datum_field_or_none(self)
+        site_code_fk = self.get_fk_for_model_field('Site', 'code')
+        self.site_code_field = self.get_field_by_mame(site_code_fk.data_field) if site_code_fk else None
+        if self.site_code_field is None:
+            self.latitude_field = self.find_latitude_field_or_throw(self)
+            self.longitude_field = self.find_longitude_field_or_throw(self)
+            self.datum_field = self.find_datum_field_or_none(self)
+        else:
```

Comment:
This seems like a convoluted way of doing things - having two different methods that either throw or don't throw. Why not just have one method but if the first case (site code is not) you throw the exception up the chain but in the second case (there is a site code) you catch the exception and do nothing with it.